### DISCUSSION TITLE
モバイル学習コンテンツでアプリ実装時、App.tsxを元に戻すことを記載しても良い

### DISCUSSION THREAD

**Discussion by nagano-ryo** [ws-4020/mobile-app-crib-notes#250] [Ideas]
学習コンテンツFBより

アプリ実装前に以下でApp.tsxを編集しているはずであり、
編集している場合はApp.tsxをプロジェクト作成直後の状態に戻す必要がある。
「App.tsxをプロジェクト作成直後の状態にしてください」や戻すべきApp.tsxの内容を記載する、などが考えられる。

https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/basic-concepts/react-native-basics
https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/basic-concepts/react-navigation-basics

・修正対象
https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/todo-app/screens/welcome

**Comment by tani-takanori** [comment]
こういう話もあります

🖋: ToDoアプリの作成でgitを使うか
https://github.com/ws-4020/mobile-app-crib-notes/discussions/106

**Comment by nagano-ryo** [comment]
手段としては以下が考えられる。
1. プロジェクト作成直後の状態のsrc/App/tsxのソースコードを記載したうえで、「以下のソースコードでsrc/App/tsxを上書きしてください。」
    - メリット
        - 実施者は楽
    - デメリット
        - 保守性が悪い（src/App/tsxに変更が入った場合、反映が漏れる） 
2. プロジェクト作成コマンドを任意の場所で再度実行しプロジェクト作成する。作成されたsrc/App/tsxで上書き。
    - メリット
        - 実施者は楽
    - デメリット
        - 特になし（ただし格好良くない） 
3. プロジェクト作成コマンド実行直後、「src/App/tsxを画面実装前に戻す必要があるので、バックアップを取得しておいてください。」
    - メリット
        - 実施者は楽
    - デメリット
        - 特になし（ただし格好良くない） 
4. git管理して戻す。
      1. プロジェクト作成の状態をコミット
          1. git init
          2. git add .
          3. git commit -m "新規作成"
      2. 画面実装開始時にsrc/App/tsxを戻す
          1. git log
          2. git reset --hard ハッシュ値

  
  
    - メリット
        - 実装していく中で修正差分が確認できる
    - デメリット
        - gitの知識が多少必要
        - 手順が増える

**Reply by tani-takanori** [reply]
ここをみてもらったら？
https://github.com/ws-4020/rn-spoiler/blob/2021.05.0/template/src/App.tsx

ここを参照してもらって戻すとアプリの生成で参照するtagが変わったときにメンテが必要になるのでdocusarus上で変数にしちゃう？

**Comment by nagano-ryo** [comment]
pull request
https://github.com/ws-4020/mobile-app-crib-notes/pull/272

**Comment by creature-water-valley** [comment]
対応済みなのでクローズ
